app/controllers/actions/IdentifierAction.scala

```diff
@@ -36,39 +38,32 @@ class AuthenticatedIdentifierAction @Inject()(
                                                config: FrontendAppConfig,
                                                val parser: BodyParsers.Default
                                              )
-                                             (implicit val executionContext: ExecutionContext) extends IdentifierAction with AuthorisedFunctions {
+                                             (implicit val executionContext: ExecutionContext)
+  extends IdentifierAction
+  with AuthorisedFunctions
+  with Logging {
 
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId => block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
-    } recover {
+    authorised(Enrolment(config.tgpEnrolmentIdentifier.key))
```

Comment:
can we ensure they're not agents as the API team have done:
https://github.com/hmrc/trader-goods-profiles/blob/main/app/uk/gov/hmrc/tradergoodsprofiles/controllers/actions/AuthAction.scala#L73

Reply:
Added

Reply:
Done